### PR TITLE
Add Travis scripts to test the plugins 

### PR DESCRIPTION
**Pull request by tmichalak**
This PR adds Travis CI for testing of the plugins.

### CODE REVIEW

**Comment by mithro** [comment]
```
Building plugins..
make: *** No rule to make target 'plugins'.  Stop.
The command "./.travis/build-and-test.sh" exited with 2.
```

**Comment by tmichalak** [comment]
Depends on #4. I separated the Travis chages from #4 PR as @litghost requested. However I see how this is a chicken-egg problem, because we want to make sure #4 is ok using Travis. 
I temporarily pushed commits from #4 here to show that Travis works. Once #4 is merged I will rebase and this PR will only have the Travis commit.

**Comment by litghost** [comment]
> Depends on #4. I separated the Travis chages from #4 PR as @litghost requested. However I see how this is a chicken-egg problem, because we want to make sure #4 is ok using Travis.
> I temporarily pushed commits from #4 here to show that Travis works. Once #4 is merged I will rebase and this PR will only have the Travis commit.

What I'd like is a simple PR that sets up an empty test that passes, we merge that, and then #4 adds the test.  Can we do that?

**Comment by tmichalak** [comment]
> What I'd like is a simple PR that sets up an empty test that passes, we merge that, and then #4 adds the test. Can we do that?

I added some dummy test and extracted the test Makefiles from #4.

File: .travis/build-and-test.sh [hunk 1]
```diff
@@ -0,0 +1,61 @@
+#! /bin/bash
+
+set -e
+
+source .travis/common.sh
+
+##########################################################################
+
+echo
```
Comment (mithro) [inline .travis/build-and-test.sh]: Can you create a new issue to replace these with calls to replace these echos with things like https://github.com/litex-hub/litex-conda-packages/blob/9b201004a699af707afc0745e57a06c223e4f545/.travis/common.sh#L32-L43 ?

**Pull request merged**